src/protocol/sort/generate_permutation.rs

```diff
@@ -98,21 +98,22 @@ pub async fn generate_permutation<F: Field>(
     num_bits: u32,
 ) -> Result<Vec<Replicated<F>>, Error> {
     let ctx_0 = ctx.narrow(&Sort(0));
-    let bit_0 =
-        convert_shares_for_a_bit(ctx_0.narrow(&ModulusConversion), input, num_bits, 0).await?;
+    let triple_0 = convert_bits_local(ctx_0.role(), 0, input);
+    let bit_0 = convert_bits(ctx_0.narrow(&ModulusConversion), &triple_0).await?;
```

Comment:
And `bit_0` sounds like just one bit... but this is a list. Let's at least use a plural if not `bit_0_list`